AST MemberExpr node has optional `non_odr_use_unevaluated` token; as used by `sizeof(x)`

The `MemberExpr` node has an optional `non_odr_use_unevaluated` token in Clang 9.0. Note, this issue is almost identical to #843 (which details the optional `non_odr_use_unevaluated` on `DeclRefExpr` nodes).

# Example

Contents of `k.c`:
```c
typedef struct T {
	int x;
} T;


int f(T t) {
	return sizeof(t.x);
}
```

# Panic

```
$ c2go transpile k.c
panic: could not match regexp with string
^(?P<address>[0-9a-fx]+) <(?P<position>.*)> '(?P<type>.*?)'(?P<type2>:'.*?')?(?P<lvalue> lvalue)?(?P<bitfield> bitfield)? (?P<pointer>[->.]+)(?P<name>\w+)? (?P<address2>[0-9a-fx]+)[\s]*$
0x556ac141bb30 <col:16, col:18> 'int' lvalue .x 0x556ac141b7c8 non_odr_use_unevaluated


goroutine 52 [running]:
github.com/elliotchance/c2go/ast.groupsFromRegex(0xc0001c23c0, 0xba, 0xc0002107a4, 0x56, 0xa)
	/home/u/goget/src/github.com/elliotchance/c2go/ast/ast.go:290 +0x360
github.com/elliotchance/c2go/ast.parseMemberExpr(0xc0002107a4, 0x56, 0x6b334d)
```

# AST output

The AST of `k.c` looks as follows:

```
$ c2go ast k.c
TranslationUnitDecl 0x55dffefac9f8 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x55dffefad290 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x55dffefacf90 '__int128'
|-TypedefDecl 0x55dffefad300 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x55dffefacfb0 'unsigned __int128'
|-TypedefDecl 0x55dffefad5e8 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x55dffefad3e0 'struct __NSConstantString_tag'
|   `-Record 0x55dffefad358 '__NSConstantString_tag'
|-TypedefDecl 0x55dffefad680 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x55dffefad640 'char *'
|   `-BuiltinType 0x55dffefaca90 'char'
|-TypedefDecl 0x55dffefad958 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x55dffefad900 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x55dffefad760 'struct __va_list_tag'
|     `-Record 0x55dffefad6d8 '__va_list_tag'
|-RecordDecl 0x55dfff00b710 </home/u/Desktop/ccc/k/k.c:1:9, line:3:1> line:1:16 struct T definition
| `-FieldDecl 0x55dfff00b7c8 <line:2:2, col:6> col:6 referenced x 'int'
|-TypedefDecl 0x55dfff00b878 <line:1:1, line:3:3> col:3 referenced T 'struct T':'struct T'
| `-ElaboratedType 0x55dfff00b820 'struct T' sugar
|   `-RecordType 0x55dfff00b790 'struct T'
|     `-Record 0x55dfff00b710 'T'
`-FunctionDecl 0x55dfff00ba20 <line:6:1, line:8:1> line:6:5 f 'int (T)'
  |-ParmVarDecl 0x55dfff00b920 <col:7, col:9> col:9 referenced t 'T':'struct T'
  `-CompoundStmt 0x55dfff00bbc8 <col:12, line:8:1>
    `-ReturnStmt 0x55dfff00bbb8 <line:7:2, col:19>
      `-ImplicitCastExpr 0x55dfff00bba0 <col:9, col:19> 'int' <IntegralCast>
        `-UnaryExprOrTypeTraitExpr 0x55dfff00bb80 <col:9, col:19> 'unsigned long' sizeof
          `-ParenExpr 0x55dfff00bb60 <col:15, col:19> 'int' lvalue
            `-MemberExpr 0x55dfff00bb30 <col:16, col:18> 'int' lvalue .x 0x55dfff00b7c8 non_odr_use_unevaluated
              `-DeclRefExpr 0x55dfff00bb10 <col:16> 'T':'struct T' lvalue ParmVar 0x55dfff00b920 't' 'T':'struct T' non_odr_use_unevaluated
```

--- a/ast/member_expr.go
+++ b/ast/member_expr.go
@@ -2,16 +2,17 @@ package ast
 
 // MemberExpr is expression.
 type MemberExpr struct {
-	Addr       Address
-	Pos        Position
-	Type       string
-	Type2      string
-	Name       string
-	IsLvalue   bool
-	IsBitfield bool
-	Address2   string
-	IsPointer  bool
-	ChildNodes []Node
+	Addr                 Address
+	Pos                  Position
+	Type                 string
+	Type2                string
+	Name                 string
+	IsLvalue             bool
+	IsBitfield           bool
+	Address2             string
+	IsPointer            bool
+	NonODRUseUnevaluated bool
+	ChildNodes           []Node
 }
 
 func parseMemberExpr(line string) *MemberExpr {
@@ -23,7 +24,8 @@ func parseMemberExpr(line string) *MemberExpr {
 		(?P<bitfield> bitfield)?
 		 (?P<pointer>[->.]+)
 		(?P<name>\w+)?
-		 (?P<address2>[0-9a-fx]+)`,
+		 (?P<address2>[0-9a-fx]+)
+		(?P<non_odr_use_unevaluated> non_odr_use_unevaluated)?`,
 		line,
 	)
 
@@ -33,16 +35,17 @@ func parseMemberExpr(line string) *MemberExpr {
 	}
 
 	return &MemberExpr{
-		Addr:       ParseAddress(groups["address"]),
-		Pos:        NewPositionFromString(groups["position"]),
-		Type:       groups["type"],
-		Type2:      type2,
-		IsPointer:  groups["pointer"] == "->",
-		Name:       groups["name"],
-		IsLvalue:   len(groups["lvalue"]) > 0,
-		IsBitfield: len(groups["bitfield"]) > 0,
-		Address2:   groups["address2"],
-		ChildNodes: []Node{},
+		Addr:                 ParseAddress(groups["address"]),
+		Pos:                  NewPositionFromString(groups["position"]),
+		Type:                 groups["type"],
+		Type2:                type2,
+		IsPointer:            groups["pointer"] == "->",
+		Name:                 groups["name"],
+		IsLvalue:             len(groups["lvalue"]) > 0,
+		IsBitfield:           len(groups["bitfield"]) > 0,
+		Address2:             groups["address2"],
+		NonODRUseUnevaluated: len(groups["non_odr_use_unevaluated"]) > 0,
+		ChildNodes:           []Node{},
 	}
 }
 

--- a/ast/member_expr_test.go
+++ b/ast/member_expr_test.go
@@ -90,6 +90,19 @@ func TestMemberExpr(t *testing.T) {
 			IsPointer:  false,
 			ChildNodes: []Node{},
 		},
+		`0x55dfff00bb30 <col:16, col:18> 'int' lvalue .x 0x55dfff00b7c8 non_odr_use_unevaluated`: &MemberExpr{
+			Addr:                 0x55dfff00bb30,
+			Pos:                  NewPositionFromString("col:16, col:18"),
+			Type:                 "int",
+			Type2:                "",
+			IsLvalue:             true,
+			IsBitfield:           false,
+			Name:                 "x",
+			Address2:             "0x55dfff00b7c8",
+			IsPointer:            false,
+			NonODRUseUnevaluated: true,
+			ChildNodes:           []Node{},
+		},
 	}
 
 	runNodeTests(t, nodes)

